ports/models.py

```diff
@@ -28,47 +34,398 @@ class Port(models.Model):
+                        domain = maintainer.get('email', {}).get('domain', '')
+                        github = maintainer.get('github', '')
+
+                        maintainer_object, created = Maintainer.objects.get_or_create(
+                            name=name,
+                            domain=domain,
+                            github=github
+                        )
+
+                        maintainer_object.ports.add(new_port)
+                except KeyError:
+                    pass
+
+                try:
+                    for variant in port['variants']:
+                        variant_object = Variant()
+                        variant_object.port = new_port
+                        variant_object.variant = variant
+                        variant_object.save()
+                except KeyError:
+                    pass
+
+        def load_dependencies_table(ports):
+
+            def load_depends(list_of_dependencies, type_of_dependency, port):
+                dependency = Dependency()
+                dependencies = []
+                dependency.type = type_of_dependency
+                dependency.port_name = port
+                for depends in list_of_dependencies:
+                    try:
+                        dependencies.append(Port.objects.get(name__iexact=depends.rsplit(':', 1)[-1]))
+                    except Port.DoesNotExist:
+                        print("Failed to append {} as a dependency to {}. Not Found.".format(depends.rsplit(':', 1)[-1],
+                                                                                             port.name))
+                dependency.save()
+                dependency.dependencies.add(*dependencies)
+
+            for port in ports:
+                try:
+                    port_object = Port.objects.get(name__iexact=port['name'])
+                    for dependency_type in ["lib", "extract", "run", "patch", "build", "test", "fetch"]:
+                        key = "depends_" + dependency_type
+                        if key in port:
+                            load_depends(port[key], dependency_type, port_object)
+
+                except Port.DoesNotExist:
+                    print("Failed to update dependencies for {}. Port not found in database.".format(port['name']))
+        if isinstance(data, list):
+            load_ports_and_maintainers_table(data)
+            load_categories_table(data)
+            load_dependencies_table(data)
+        elif isinstance(data, str):
+            ports = open_portindex_json(data)
+            load_categories_table(ports)
+            load_ports_and_maintainers_table(ports)
+            load_dependencies_table(ports)
+
+    @classmethod
+    def update(cls, data, is_json=True):
+        def open_portindex_json(path):
+            with open(path, "r") as file:
+                ports = json.load(file)
+            return ports
+
+        def open_ports_from_list(list_of_ports, path='portindex.json'):
```

Comment:
May I know the use case for this function?

Reply:
The method of using commit hash would give us a list of ports that need to be updated. This function would fetch those ports from PortIndex.json and return a list of JSON objects which can then be passed to `run_updates`.

Reply:
Okay, thank you. I do not know the format of `portindex.json` but in order to avoid any delay in me going through the code of `portindex2json.tcl`, I'd request you to check if it's possible to iterate over `list_of_ports` instead and see if a port can be accessed in O(1) from `portindex.json` (which highly depends on how the ports are stored in portindex.json)?

Reply:
The format of `portindex.json`  is a list of JSON objects. After looking into many options, I will the best week can do is list comprehensions.

Here are the formats:
```
list_of_ports = ["portA", "portB", ..... ]
```

portindex.json
```
[
  {
    "name": "portA",
    "version": 1,
    "random_key": "value"
    .
    .
  },
  {
    "name": "portB",
    "version": 2,
    "random_key": "value2"
    .
    .
  },
...
...
...
}]

Reply:
The format of portindex.json output doesn't directly allow us to access a port by its name. While it can still be improved, I'm anyway resolving this conversation because the feature is not likely to be used much I think and we can always improve it later when we incur any delay issues.

Reply:
Carried to #30